File: 0169-majority-element/0169-majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        count = 0
        candidate = None

        for num in nums:
            if count == 0:
                candidate = num
            if num == candidate:
                count += 1
            else:
                count -= 1

        # At this point, candidate contains a potential majority element
        # We need to check if it appears more than n/2 times in the list

        count = 0
        for num in nums:
            if num == candidate:
                count += 1

        if count > len(nums) / 2:
            return candidate
        else:
            return -1

File: 0169-majority-element/0169-majority-element.cs
public class Solution {
    public int MajorityElement(int[] nums)
{
    int candidate = nums[0];
    int count = 1;

    for (int i = 1; i < nums.Length; i++)
    {
        if (count == 0)
        {
            candidate = nums[i];
            count = 1;
        }
        else if (nums[i] == candidate)
        {
            count++;
        }
        else
        {
            count--;
        }
    }

    if (count > 0 && nums.Count(x => x == candidate) > nums.Length / 2)
    {
        return candidate;
    }
    else
    {
        return -1;
    }
}
}